specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+- An attribute indicating the total number of qubits used by the program
+- An attribute indicating the total number of classical bits required throughout the program to store measurement outcomes
+
+These attributes will show up as an [attribute group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by multiple function definitions or global variables. 
+Arbitrary custom attributes may be optionally attached to any of the declared functions to convey additional information about that function. Consumers of Base Profile compliant programs should hence not rely on the numbering of the entry point attribute group, but instead look for function to which an attribute with the name `entry_point` is attached to determine which one to invoke when the program is launched.
+
+To indicate the total number of qubits required to execute the program, a custom attribute with the name `required_qubits` is defined and attached to the entry point. To indicate the number of registers/bits needed to store measurement results during program execution, a custom attribute with the name `required_results` is defined and attached to the entry point. The value of both of these attributes is the string representation of a 64-bit integer constant.
+
+TODO: why the string value, and not an integer?   
+TODO: no double underscore guard for attributes likely makes sense
+
+### Function Body
+
+The function body consists of a single block named `entry`. This implies in particular that no branching is permitted inside a function's body.
+In the `entry` block, any number of calls to QIS functions may be performed.
+To be compatible with the Base Profile these functions must return void. 
+Any arguments to invoke them must be inlined into the call itself; 
+they hence must be constants or a pointer to a [qubit or result]()
+The result of the program execution should be logged using output recording functions.
+
+The following instructions are the *only* LLVM instructions that are permitted within a Base Profile compliant program:
+
+| LLVM Instruction                  | Context and Purpose         | Rules for Usage |
+|---------------------------|-------------------|-------------|
+| `call`    | Used within a function block to invoke any one of the declared QIS functions and the output recording functions. | May optionally be preceded by a [`tail` marker](https://llvm.org/docs/LangRef.html#call-instruction). |
+| `ret` | Used to return the exit code of the program. | Must occur (only) as the final instruction of the `entry` block. |
+| `inttoptr` | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`. | May be used as part of a function call only. |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value. | May be used as part of call to an output recording function only. |
```

Comment:
`getelementptr` is not supported by any of the [current profiles](https://github.com/qir-alliance/qat/blob/main/qir/qat/ValidationPass/ValidationPassConfiguration.cpp) neither is pointer type `i8*`.

Reply:
See my reply on another conversation.